src/System.Management.Automation/utils/ResourceManagerCache.cs

```diff
@@ -177,8 +177,37 @@ internal static bool DFT_DoMonitorFailingResourceLookup
                 throw PSTraceSource.NewArgumentException("resourceId");
             }
 
-            ResourceManager resourceManager = GetResourceManager(assembly, baseName);
-            string text = resourceManager.GetString(resourceId);
+            ResourceManager resourceManager = null;
+            string text = string.Empty;
+
+            // FYI: for a non-existing resource defined by {assembly,baseName,resourceId}
+            // MissingManifestResourceException is thrown only at the time when resource retrieval method such as ResourceManager.GetString or ResourceManager.GetObject is called,
+            // Not when you instantiate a ResourceManager object.
+            try
+            {
+                // try with original baseName
+                resourceManager = GetResourceManager(assembly, baseName);
+                text = resourceManager.GetString(resourceId);
+            }
+            catch (MissingManifestResourceException)
+            {
+                // original baseName failed;  lets try with alternative resource path format
+                const string resourcesSubstring = ".resources.";
+                int resourcesSubstringIndex = baseName.IndexOf(resourcesSubstring);
+                string newBaseName = string.Empty;
+                if (resourcesSubstringIndex != -1)
+                {
+                    newBaseName = baseName.Substring(resourcesSubstringIndex + resourcesSubstring.Length); // e.g.  "FileSystemProviderStrings"
+                }
+                else
+                {
+                    newBaseName = assembly.GetName().Name + resourcesSubstring + baseName; // e.g. "System.Management.Automation.resources.FileSystemProviderStrings"

```

Comment:
A single call to `string.Concat` with 4 or fewer arguments could be better than using a `StringBuilder` - the length of the string can be pre-calculated, so there would only be 1 allocation instead of multiple for the `StringBuilder`, the array of characters in `StringBuilder`, and the final string allocation.